substrate-node/pallets/pallet-smart-contract/src/contract_migration.rs

```diff
@@ -120,6 +120,15 @@ pub fn migrate_to_version_4<T: Config>() -> frame_support::weights::Weight {
         let mut migrated_count = 0;
         // We transform the storage values from the old into the new format.
         Contracts::<T>::translate::<deprecated::ContractV3, _>(|k, ctr| {
+            // Don't map deleted contracts
+            if matches!(ctr.state, types::ContractState::Deleted(_)) {
+                info!(
+                    "contract with id: {:?} in state delete, skipping...",
+                    ctr.contract_id
+                );
+                return None;
+            }
+
```

Comment:
old contracts should be mapped as well if they are still in storage, otherwise they won't be able to be decoded

Reply:
? in the new code all contracts in Delete state are also removed from chain. Making the `Deleted` state deprecated

Reply:
`Contracts` storage map is a list with Optional Contract objects. So returning `None` is fine